packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,384 @@
+        logger.info(f"Initialized the HTTP Server on HOST: {host} and PORT: {port}")
+
+    def _process_api(self, api_spec):
+        """Process the api_spec.
+
+        Decode all paths to get envelops each, and put it in the agent's inbox.
+        """
+        try:
+            paths = api_spec["paths"]
+        except KeyError:
+            logger.error("Key 'paths' not found in API spec.")
+
+        for path_name, path_value in paths.items():
+            try:
+                envelope = self._decode_path(path_name, path_value)
+                assert self.in_queue is not None, "Input queue not initialized."
+                assert self.loop is not None, "Loop not initialized."
+                # Ok to put these Envelopes in_queue during HTTPChannel initialization?
+                asyncio.run_coroutine_threadsafe(self.in_queue.put(envelope), self.loop)
+            except ValueError:
+                logger.error(f"Bad formatted path: {path_name}")
+            except Exception as e:
+                logger.error(f"Error when processing a path. Message: {str(e)}")
+
+    def _decode_path(self, path_name: str, path_value: dict, separator: str = "/"):
+        """Path --> Envelope
+
+        Convert specified API path name and value into in_queue'd Envelope.
+        Envelope's msg consists of the response options for each REST req in path.
+        """
+        path_split = path_name.split(separator)
+
+        to = path_split[1].strip()  # to: name of Skill (petstoresim)?
+        sender = "TBD"  # hardcoded for now
+        protocol_id = PublicId.from_str("fetchai/http:0.1.0")  # hardcoded for now
+        uri = URI(f"http://{self.host}:{self.port}{path_name}")
+        context = EnvelopeContext(connection_id=self.connection_id, uri=uri)
+        # Msg consists of all of the responses for each REST req in JSON format.
+        responses = {}
+        for req_type, req_value in path_value.items():
+            responses[req_type] = req_value["responses"]
+        msg_bytes = json.dumps(responses).encode()
+
+        return Envelope(
+            to=to,
+            sender=sender,
+            protocol_id=protocol_id,
+            message=msg_bytes,
+            context=context,
+        )
+
+    def _process_request(
+        self, http_method: str, url: str, param: str, body: str = "{}"
+    ):
+        """Process incoming API request by packaging into Envelope and sending it in-queue.
+
+        """
+        self._client_id = uuid4().hex
+        protocol_id = PublicId.from_str("fetchai/http:0.1.0")
+        uri = URI(f"http://{self.host}:{self.port}{url}")
+        context = EnvelopeContext(connection_id=self.connection_id, uri=uri)
+        # Prepare the Envelope's message body and encode it into bytes.
+        msg = {
+            "performative": http_method,
+            "path": url,
+            "params": param,
+            "payload": body,
+        }
+        msg_bytes = json.dumps(msg).encode()
```

Comment:
This is ok for now, but we will need to create an actual protocol to represent these messages once the other issues are resolved.

Reply:
indeed, next iteration or so